content/docs/user-guide/dvc-files-and-directories.md

```diff
@@ -1,8 +1,152 @@
+
+  - `url`: URL of Git repository with source DVC project
+  - `rev`: Only present when the `--rev` option of `dvc import` is used.
+    Specific commit hash, branch or tag name, etc. (a
+    [Git revision](https://git-scm.com/docs/revisions)) used to import the
+    dependency from.
+  - `rev_lock`: Git commit hash of the external <abbr>DVC repository</abbr> at
+    the time of importing or updating (with `dvc update`) the dependency.
+
+Note that comments can be added to `.dvc` files and `dvc.yaml` using the
+`# comment` syntax.
+
+> `meta` fields and `#` comments are preserved among executions of the
+> `dvc repro` and `dvc commit` commands, but not when a `.dvc` file is
+> overwritten by `dvc add`,`dvc import`, or `dvc import-url`.
+
+## dvc.yaml files
+
+When you add commands to a pipeline with `dvc run`, the `dvc.yaml` file is
+created or updated. Here's a simple example:
+
+```yaml
+stages:
+  features:
+    cmd: jupyter nbconvert --execute featurize.ipynb
+    deps:
+      - data/clean
+    params:
+      - levels.no
+    outs:
+      - features
+    metrics:
+      - performance.json
+  training:
+    cmd: python train.py
+    deps:
+      - train.py
+      - features
+    outs:
+      - model.pkl
+    plots:
+      - logs.csv:
+          x: epoch
+          x_label: Epoch
+    meta: 'For deployment'
+    # User metadata and comments are supported.
+```
+
+`dvc.yaml` files consists of a group of `stages` with names provided explicitly
+by the user with the `--name` (`-n`) option of `dvc run`.
+
+Each stage's contents are similar to individual [`dvc` files](#dvcfiles) but
+they can contain more information in `dvc.yaml` These are the possible following
+fields:
+
+- `cmd`: Executable command defined in this stage
+- `deps` (optional): List of <abbr>dependency</abbr> file or directory paths of
+  this stage
+- `params` (optional): List of the [parameters](/doc/command-reference/params).
+  These are key paths referring to another YAML file (`params.yaml` by default).
+- `outs` (optional): List of <abbr>output</abbr> file or directory paths of this
+  stage
+- `metrics` (optional): List of [metric files](/doc/command-reference/metrics)
+- `plots` (optional): List of [plot metrics](/doc/command-reference/plots) and
+  optionally, their default configuration (subfields matching the options of
+  `dvc plots modify`).
+- `frozen` (optional): Whether or not this stage is frozen from reproduction
+- `always_changed` (optional): Whether or not this stage is considered as
+  changed by commands such as `dvc status` and `dvc repro`. `false` by default
+- `meta` (optional): Arbitrary metadata can be added manually with this field.
+  Any YAML contents is supported. `meta` contents are ignored by DVC, but they
+  can be meaningful for user processes that read `.dvc` files.
+
+`dvc.yaml` files also support `# comments`.
+
+> `meta` fields and `#` comments are always preserved in `dvc.yaml` stages.
```

Comment:
are they preserved if I overwrite the stage? cc @skshetry @efiop ?

Reply:
@jorgeorpinel no need to put it as a note - it can be part of the previous paragraph(s)

Reply:
also, if they are preserved I would omit it completely - this what anyone would expect anyway - anythings else is a bug that is not fixed yet in DVC

Reply:
that we should mention about

Reply:
Yes, they are preserved when overwriting too (I don't know if that's what we want, but when implementing the "maintain the order when overwrite" I cut the corner and did it the fastest/easiest way).

Reply:
I agree with @shcheklein  and I have mentioned this already somewhere. There's no need for the note at all.

Reply:
> if they are preserved I would omit it completely - this what anyone would expect anyway

Saugat also mentioned this earlier, same answer:  for .dvc files the behavior is different and has a note about it (making both notes part of the paragraph now though) so if you read that first in the .dvc file section you may expect that dvc.yaml files behave similarly, thus we need to note here as well, I think.